pkg/alertmanager/types.go

```diff
@@ -28,12 +28,12 @@ import (
 // marshalling. See the following issue for details:
 // https://github.com/prometheus/alertmanager/issues/1985
 type alertmanagerConfig struct {
-	Global            *globalConfig      `yaml:"global,omitempty" json:"global,omitempty"`
-	Route             *route             `yaml:"route,omitempty" json:"route,omitempty"`
-	InhibitRules      []*inhibitRule     `yaml:"inhibit_rules,omitempty" json:"inhibit_rules,omitempty"`
-	Receivers         []*receiver        `yaml:"receivers,omitempty" json:"receivers,omitempty"`
-	MuteTimeIntervals []muteTimeInterval `yaml:"mute_time_intervals,omitempty" json:"mute_time_intervals,omitempty"`
-	Templates         []string           `yaml:"templates" json:"templates"`
+	Global            *globalConfig       `yaml:"global,omitempty" json:"global,omitempty"`
+	Route             *route              `yaml:"route,omitempty" json:"route,omitempty"`
+	InhibitRules      []*inhibitRule      `yaml:"inhibit_rules,omitempty" json:"inhibit_rules,omitempty"`
+	Receivers         []*receiver         `yaml:"receivers,omitempty" json:"receivers,omitempty"`
+	MuteTimeIntervals []*muteTimeInterval `yaml:"mute_time_intervals,omitempty" json:"mute_time_intervals,omitempty"`
```

Comment:
Why changing to a slice of pointers?

Reply:
The main reason was that it matched the format of the other fields. Secondly I thought it might be better to have nil value than default zero values for ints that might lead to some confusion down the road.

Reply:
right I missed that other fields were already slices of pointers :)